pyvisa_py/tcpip.py

```diff
@@ -64,8 +71,48 @@ class TCPIPInstrSession(Session):
 
     @staticmethod
     def list_resources() -> List[str]:
-        # TODO: is there a way to get this?
-        return []
+        broadcast_addr = []
+        if psutil is not None:
+            # Get broadcast address for each interface
+            for interface, snics in psutil.net_if_addrs().items():
+                for snic in snics:
+                    if snic.family is socket.AF_INET:
+                        addr = snic.address
+                        mask = snic.netmask
+                        network = ipaddress.IPv4Network(addr + "/" + mask, strict=False)
+                        broadcast_addr.append(str(network.broadcast_address))
+        else:
+            msg = "To discover devices on all interfaces, please install psutil"
+            Session.register_unavailable(constants.InterfaceType.tcpip, "INSTR", msg)
+            broadcast_addr.append("255.255.255.255")
+
+        try:
+            pmap_list = [rpc.BroadcastUDPPortMapperClient(ip) for ip in broadcast_addr]
+            for pmap in pmap_list:
+                pmap.set_timeout(0)
+                pmap.send_port(
+                    (vxi11.DEVICE_CORE_PROG, vxi11.DEVICE_CORE_VERS, rpc.IPPROTO_TCP, 0)
+                )
+
+            # Timeout for responses
+            time.sleep(1)
+
+            all_res = []
+            for pmap in pmap_list:
+                resp = pmap.recv_port(
+                    (vxi11.DEVICE_CORE_PROG, vxi11.DEVICE_CORE_VERS, rpc.IPPROTO_TCP, 0)
+                )
+                res = [r[1][0] for r in resp if r[0] > 0]
+                res = sorted(
+                    res, key=lambda ip: tuple(int(part) for part in ip.split("."))
+                )
+                # TODO: Detect GPIB over TCPIP
+                res = ["TCPIP::{}::INSTR".format(host) for host in res]
+                all_res.extend(res)
+        except rpc.RPCError:
```

Comment:
On Windows I get a crash with an os error (OSError: [WinError 10051] A socket operation was attempted to an unreachable network). Need to be fixed either in rpc.py or here. rpc.py may make more sense